.github/scripts/generate_ci_workflows.py

```diff
@@ -143,13 +147,22 @@ def assert_valid(self) -> None:
             assert self.test_runner_type in LINUX_RUNNERS, err_message
         if self.arch == 'windows':
             assert self.test_runner_type in WINDOWS_RUNNERS, err_message
+    
+    def generate_workflow_name(self, workflow) -> str:
+        workflow_name = workflow.build_environment
+        if not workflow.on_pull_request:
```

Comment:
Maybe at the point where we start to add things like this then inheritance becomes the answer.

I really want these to be more explicit rather than taken care of in conditional logic somewhere

Reply:
yeah I thought about it as well but it makes it hard to mix and match --> like if it is a periodic / master-only job.

Reply:
I think this is better just set the workflow_name individually for each workflow, the benefit is that this conditional logic may not hold for future requirements and we're free to change that for each one separately, and even inheritance doesn't solve the it (it just split them into multiple subclasses).

Reply:
yes. i agree to some extend on this statement as well . that's probably why I kept the linux version, py, cuda, and gcc versions non-generated. cuz they might not be a "relevant" argument universally. the only ones that I thought will be universal is "master-only" and "periodic", which are essentially not affecting any build/test steps, but just affect the CI timing. 

But yeah regardless I think for now this approach is good just for the 2?